### ISSUE TITLE
Let helm-occur work on regions

### ISSUE THREAD

**Issue by Ambrevar** [emacs-helm/helm#2019]
At the moment helm-occur only works over the entire buffer.  When a region is active, helm-occur still searches the entire buffer.

I think it would be nice to include an option so that helm-occur automatically restricts itself to the active region.

Note that a similar result can be obtained by narrowing the buffer to the region.  It's a bit less practical in my opinion.

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> At the moment helm-occur only works over the entire buffer. When a region is active, helm-occur still searches the entire buffer.
>
> I think it would be nice to include an option so that helm-occur automatically restricts itself to the active region.

It's easy to do:

    (defun helm-occur-on-region ()
      (interactive)
      (if (use-region-p)
          (save-restriction
            (narrow-to-region (region-beginning) (region-end))
            (deactivate-mark t)
            (call-interactively 'helm-occur))
        (call-interactively 'helm-occur)))

> Note that a similar result can be obtained by narrowing the buffer to
> the region. It's a bit less practical in my opinion.

Not when done automatically like above.

--
Thierry

**Comment by Ambrevar** [comment]
Indeed, but would you consider merging that into the current definition
of `helm-occur`?  I think such a feature is the kind of behaviours the user would expect.

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> Indeed, but would you consider merging that into the current
> definition of `helm-occur`? I think such a feature is the kind of
> behaviours the user would expect.

Will do once the resume problem fixed.

Thanks.

--
Thierry

**Comment by thierryvolpiatto** [comment]
Pierre Neidhardt <notifications@github.com> writes:

> Indeed, but would you consider merging that into the current definition
> of `helm-occur`?

Done, thanks.

-- 
Thierry
